toucan_connectors/http_api/pagination_configs.py

```diff
@@ -0,0 +1,128 @@
+from enum import Enum
+from typing import Any, Optional
+
+from pydantic import BaseModel, Field
+
+from toucan_connectors.common import FilterSchema
+from toucan_connectors.http_api.http_api_data_souce import HttpAPIDataSource
+
+
+class PaginationConfig(BaseModel):
+    """Base class for pagination configs.
+
+    Config applied for connectors without pagination configured.
+    Useful for connectors that can return all results at once.
+    """
+
+    def apply_pagination_to_data_source(self, data_source: HttpAPIDataSource) -> HttpAPIDataSource:
+        """Apply pagination to data source and return it"""
+        return data_source
+
+    def get_next_pagination_config(self, result: Any, pagination_info: Any | None) -> Optional["PaginationConfig"]:
+        """Computes next pagination config based on the parsed API responses"""
+        return None
+
+    def get_pagination_info_filter(self) -> str | None:
+        """Returns the JQ filter that must be applied to the raw API result to retrieve pagination info"""
+        return None
+
+
+class OffsetLimitPaginationConfig(PaginationConfig):
+    offset_name: str = "offset"
+    offset: int = 0
+    limit_name: str = "limit"
+    limit: int
+
+    def apply_pagination_to_data_source(self, data_source: HttpAPIDataSource) -> HttpAPIDataSource:
+        offset_limit_params = {self.offset_name: self.offset, self.limit_name: self.limit}
+        if data_source.params is None:
+            data_source.params = offset_limit_params
+        else:
+            data_source.params |= offset_limit_params
+        return data_source
```

Comment:
Implem looks good, but I'd prefer if it did not mutate the passed data source:

```suggestion
    def apply_pagination_to_data_source(self, data_source: HttpAPIDataSource) -> HttpAPIDataSource:
        offset_limit_params = {self.offset_name: self.offset, self.limit_name: self.limit}
        if data_source.params is None:
            params = offset_limit_params
        else:
            params |= offset_limit_params
        return data_source.model_copy(updates={"params": params})
```